### ISSUE TITLE
Failed to start: Unable to listen to required ports

### ISSUE THREAD

**Issue by BobMali** [ddev/ddev#5189]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
Running bash [-c /tmp/test_ddev.sh] 
======= Existing project config =========
These config files were loaded for project app: [/home/***/projects/app/typo3/.ddev/config.yaml]
name: app
type: typo3
docroot: Source/web
php_version: 7.4
webserver_type: nginx-fpm
webimage: drud/ddev-webserver:v1.21.5
router_http_port: 80
router_https_port: 443
xdebug_enabled: true
additional_hostnames: []
additional_fqdns: []
database: {mariadb 10.5}
omit_containers: [dba]
host_db_port: 32681
mailhog_port: 8025
mailhog_https_port: 8026
phpmyadmin_port: 8036
phpmyadmin_https_port: 8037
project_tld: ddev.site
use_dns_when_possible: true
composer_root: Source
composer_version: 2
disable_settings_management: true
nodejs_version: 16
default_container_timeout: 120
======= Creating dummy project named  tryddevproject-8285 in ../tryddevproject-8285 =========
OS Information: Linux laptopc225 5.4.0-153-generic #170-Ubuntu SMP Fri Jun 16 13:43:31 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
User information: uid=1542(***) gid=100(users) groups=100(users),27(sudo),121(docker)
DDEV version:  ITEM             VALUE                                   
 DDEV version     v1.21.6                                 
 architecture     amd64                                   
 db               drud/ddev-dbserver-mariadb-10.4:v1.21.5 
 dba              phpmyadmin:5                            
 ddev-ssh-agent   drud/ddev-ssh-agent:v1.21.5             
 docker           24.0.4                                  
 docker-compose   v2.15.1                                 
 docker-platform  laptopc225                              
 mutagen          0.16.0                                  
 os               linux                                   
 router           drud/ddev-router:v1.21.5                
 web              drud/ddev-webserver:v1.21.5             
PROXY settings: HTTP_PROXY='' HTTPS_PROXY='' http_proxy='' NO_PROXY=''
======= DDEV global info =========
Global configuration:
instrumentation-opt-in=false
omit-containers=[]
mutagen-enabled=false
nfs-mount-enabled=false
router-bind-all-interfaces=false
internet-detection-timeout-ms=3000
disable-http2=false
use-letsencrypt=false
letsencrypt-email=
table-style=default
simple-formatting=false
auto-restart-containers=false
use-hardened-images=false
fail-on-hook-fail=false
required-docker-compose-version=
use-docker-compose-from-path=false
project-tld=
xdebug-ide-location=
no-bind-mounts=false
use-traefik=false
wsl2-no-windows-hosts-mgt=false

======= DOCKER info =========
docker location: -rwxr-xr-x 1 root root 35927800 Jul  7 16:50 /usr/bin/docker
docker version: 
Client: Docker Engine - Community
 Version:           24.0.4
 API version:       1.43
 Go version:        go1.20.5
 Git commit:        3713ee1
 Built:             Fri Jul  7 14:50:57 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          24.0.4
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.5
  Git commit:       4ffc614
  Built:            Fri Jul  7 14:50:57 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
DOCKER_DEFAULT_PLATFORM=notset
======= Mutagen Info =========
======= Docker Info =========
Docker platform: laptopc225 
Using docker context: default (unix:///var/run/docker.sock) 
docker-compose: v2.15.1 
Using DOCKER_HOST=unix:///var/run/docker.sock 
Docker version: 24.0.4 
Able to run simple container that mounts a volume. 
Able to use internet inside container. 
Docker disk space:
Filesystem                Size      Used Available Use% Mounted on
overlay                 467.9G    198.4G    245.7G  45% /

The ddev-ssh-agent container has been removed. When you start it again you will have to use 'ddev auth ssh' to provide key authentication again.
Existing docker containers: 
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
Creating a new ddev project config in the current directory (/home/***/projects/app/tryddevproject-8285) 
Once completed, your configuration will be written to /home/***/projects/app/tryddevproject-8285/.ddev/config.yaml
 
Configuring unrecognized codebase as project type 'php' at /home/***/projects/app/tryddevproject-8285/web 
Configuration complete. You may now run 'ddev start'. 
Network ddev_default created 
Starting tryddevproject-8285... 
Container ddev-ssh-agent  Started 
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys. 
The hostname tryddevproject-8285.ddev.site is not currently resolvable, trying to add it to the hosts file 
ddev needs to run with administrative privileges.
You may be required to enter your password for sudo or allow escalation. ddev is about to issue the command:
  sudo --preserve-env=HOME /usr/bin/ddev hostname tryddevproject-8285.ddev.site 127.0.0.1
 
Not populating custom commands or hostadditions because running with root privileges
 
Network ddev-tryddevproject-8285_default  Created 
Container ddev-tryddevproject-8285-db  Started 
Container ddev-tryddevproject-8285-dba  Started 
Container ddev-tryddevproject-8285-web  Started 
Failed to start tryddevproject-8285: Unable to listen on required ports, port 443 is already in use,
Troubleshooting suggestions at https://ddev.readthedocs.io/en/stable/users/basics/troubleshooting/#unable-listen 
┌─────────────────────┬─────────┬────────────────────────────────────┬───────────────────────────────────────┬───────────┐
│ NAME                │ STATUS  │ LOCATION                           │ URL                                   │ TYPE      │
├─────────────────────┼─────────┼────────────────────────────────────┼───────────────────────────────────────┼───────────┤
│ app                 │ stopped │ ~/projects/app/typo3               │                                       │ typo3     │
├─────────────────────┼─────────┼────────────────────────────────────┼───────────────────────────────────────┼───────────┤
│ tryddevproject-8285 │ OK      │ ~/projects/app/tryddevproject-8285 │ https://tryddevproject-8285.ddev.site │ php       │
├─────────────────────┼─────────┼────────────────────────────────────┼───────────────────────────────────────┼───────────┤
│ Router              │ stopped │ ~/.ddev                            │                                       │ original: │
│                     │         │                                    │                                       │  v1.21.5  │
└─────────────────────┴─────────┴────────────────────────────────────┴───────────────────────────────────────┴───────────┘

┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ Project: tryddevproject-8285 ~/projects/app/tryddevproject-8285 https://tryddevproject-8285.ddev.site                         │
│ Docker provider: docker 24.0.4                                                                                                │
│ Router: traditional                                                                                                           │
├────────────┬──────┬──────────────────────────────────────────────────────────────────────────────────────┬────────────────────┤
│ SERVICE    │ STAT │ URL/PORT                                                                             │ INFO               │
├────────────┼──────┼──────────────────────────────────────────────────────────────────────────────────────┼────────────────────┤
│ web        │ OK   │ https://tryddevproject-8285.ddev.site                                                │ php PHP8.0         │
│            │      │ InDocker: web:443,80,8025                                                            │ nginx-fpm          │
│            │      │ Host: 127.0.0.1:32811,32812                                                          │ docroot:'web'      │
│            │      │                                                                                      │ NodeJS:16          │
├────────────┼──────┼──────────────────────────────────────────────────────────────────────────────────────┼────────────────────┤
│ db         │ OK   │ InDocker: db:3306                                                                    │ mariadb:10.4       │
│            │      │ Host: 127.0.0.1:32808                                                                │ User/Pass: 'db/db' │
│            │      │                                                                                      │ or 'root/root'     │
├────────────┼──────┼──────────────────────────────────────────────────────────────────────────────────────┼────────────────────┤
│ PHPMyAdmin │ OK   │ https://tryddevproject-8285.ddev.site:8037                                           │                    │
│            │      │ InDocker: dba:80,80                                                                  │                    │
│            │      │ `ddev launch -p`                                                                     │                    │
├────────────┼──────┼──────────────────────────────────────────────────────────────────────────────────────┼────────────────────┤
│ Mailhog    │      │ MailHog: https://tryddevproject-8285.ddev.site:8026                                  │                    │
│            │      │ `ddev launch -m`                                                                     │                    │
├────────────┼──────┼──────────────────────────────────────────────────────────────────────────────────────┼────────────────────┤
│ All URLs   │      │ https://tryddevproject-8285.ddev.site, https://127.0.0.1:32811,                      │                    │
│            │      │ http://tryddevproject-8285.ddev.site, http://127.0.0.1:32812                         │                    │
└────────────┴──────┴──────────────────────────────────────────────────────────────────────────────────────┴────────────────────┘
 
============= ddev-tryddevproject-8285-web healtcheck run =========
+ for f in /etc/bashrc/*.bashrc
+ source /etc/bashrc/commandline-addons.bashrc
++ export 'PATH=~/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/var/www/html/vendor/bin:/var/www/html/bin:/var/www/html/vendor/bin'
++ PATH='~/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/var/www/html/vendor/bin:/var/www/html/bin:/var/www/html/vendor/bin'
++ '[' -s /home/***/.nvm/nvm.sh ']'
++ source /home/***/.nvm/nvm.sh
+++ NVM_SCRIPT_SOURCE=']'
+++ '[' -z '' ']'
+++ export NVM_CD_FLAGS=
+++ NVM_CD_FLAGS=
+++ nvm_is_zsh
+++ '[' -n '' ']'
+++ '[' -z /home/***/.nvm ']'
+++ case $NVM_DIR in
+++ unset NVM_SCRIPT_SOURCE
+++ nvm_process_parameters
+++ local NVM_AUTO_MODE
+++ NVM_AUTO_MODE=use
+++ '[' 0 -ne 0 ']'
+++ nvm_auto use
+++ local NVM_MODE
+++ NVM_MODE=use
+++ local VERSION
+++ local NVM_CURRENT
+++ '[' _use = _install ']'
+++ '[' _use = _use ']'
++++ nvm_ls_current
++++ local NVM_LS_CURRENT_NODE_PATH
+++++ command which node
++++ NVM_LS_CURRENT_NODE_PATH=/usr/bin/node
+++++ nvm_version_dir iojs
+++++ local NVM_WHICH_DIR
+++++ NVM_WHICH_DIR=iojs
+++++ '[' -z iojs ']'
+++++ '[' iojs = new ']'
+++++ '[' _iojs = _iojs ']'
+++++ nvm_echo /home/***/.nvm/versions/io.js
+++++ command printf '%s\n' /home/***/.nvm/versions/io.js
++++ nvm_tree_contains_path /home/***/.nvm/versions/io.js /usr/bin/node
++++ local tree
++++ tree=/home/***/.nvm/versions/io.js
++++ local node_path
++++ node_path=/usr/bin/node
++++ '[' @/home/***/.nvm/versions/io.js@ = @@ ']'
++++ '[' @/usr/bin/node@ = @@ ']'
++++ local previous_pathdir
++++ previous_pathdir=/usr/bin/node
++++ local pathdir
+++++ dirname /usr/bin/node
++++ pathdir=/usr/bin
++++ '[' /usr/bin '!=' '' ']'
++++ '[' /usr/bin '!=' . ']'
++++ '[' /usr/bin '!=' / ']'
++++ '[' /usr/bin '!=' /home/***/.nvm/versions/io.js ']'
++++ '[' /usr/bin '!=' /usr/bin/node ']'
++++ previous_pathdir=/usr/bin
+++++ dirname /usr/bin
++++ pathdir=/usr
++++ '[' /usr '!=' '' ']'
++++ '[' /usr '!=' . ']'
++++ '[' /usr '!=' / ']'
++++ '[' /usr '!=' /home/***/.nvm/versions/io.js ']'
++++ '[' /usr '!=' /usr/bin ']'
++++ previous_pathdir=/usr
+++++ dirname /usr
++++ pathdir=/
++++ '[' / '!=' '' ']'
++++ '[' / '!=' . ']'
++++ '[' / '!=' / ']'
++++ '[' / = /home/***/.nvm/versions/io.js ']'
++++ nvm_tree_contains_path /home/***/.nvm /usr/bin/node
++++ local tree
++++ tree=/home/***/.nvm
++++ local node_path
++++ node_path=/usr/bin/node
++++ '[' @/home/***/.nvm@ = @@ ']'
++++ '[' @/usr/bin/node@ = @@ ']'
++++ local previous_pathdir
++++ previous_pathdir=/usr/bin/node
++++ local pathdir
+++++ dirname /usr/bin/node
++++ pathdir=/usr/bin
++++ '[' /usr/bin '!=' '' ']'
++++ '[' /usr/bin '!=' . ']'
++++ '[' /usr/bin '!=' / ']'
++++ '[' /usr/bin '!=' /home/***/.nvm ']'
++++ '[' /usr/bin '!=' /usr/bin/node ']'
++++ previous_pathdir=/usr/bin
+++++ dirname /usr/bin
++++ pathdir=/usr
++++ '[' /usr '!=' '' ']'
++++ '[' /usr '!=' . ']'
++++ '[' /usr '!=' / ']'
++++ '[' /usr '!=' /home/***/.nvm ']'
++++ '[' /usr '!=' /usr/bin ']'
++++ previous_pathdir=/usr
+++++ dirname /usr
++++ pathdir=/
++++ '[' / '!=' '' ']'
++++ '[' / '!=' . ']'
++++ '[' / '!=' / ']'
++++ '[' / = /home/***/.nvm ']'
++++ nvm_echo system
++++ command printf '%s\n' system
+++ NVM_CURRENT=system
+++ '[' _system = _none ']'
+++ '[' _system = _system ']'
++++ nvm_resolve_local_alias default
++++ nvm_echo
++++ command printf '%s\n' ''
+++ VERSION=
+++ '[' -n '' ']'
+++ nvm_rc_version
++ '[' -s /home/***/.nvm/bash_completion ']'
++ source /home/***/.nvm/bash_completion
+++ command -v nvm
+++ [[ -n '' ]]
+++ complete -o default -F __nvm nvm
++ ls '/home/***/.bashrc.d/*'
bash: rm -f /tmp/healthy && /healthcheck.sh: No such file or directory
============= ddev logs =========
2023-07-24 09:45:15,875 INFO Included extra file "/etc/supervisor/php-fpm.conf" during parsing
2023-07-24 09:45:15,875 INFO Included extra file "/etc/supervisor/conf.d/mailhog.conf" during parsing
2023-07-24 09:45:15,875 INFO Included extra file "/etc/supervisor/conf.d/supervisor.conf" during parsing
2023-07-24 09:45:15,897 INFO RPC interface 'supervisor' initialized
2023-07-24 09:45:15,897 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2023-07-24 09:45:15,898 INFO supervisord started with pid 1
2023-07-24 09:45:16,900 INFO spawned: 'child_exit_monitor' with pid 2068
2023-07-24 09:45:16,902 INFO spawned: 'php-fpm' with pid 2069
2023-07-24 09:45:16,904 INFO spawned: 'nginx' with pid 2070
2023-07-24 09:45:16,906 INFO spawned: 'mailhog' with pid 2071
2023/07/24 09:45:17 [crit] 2072#2072: *1 connect() to unix:/run/php-fpm.sock failed (2: No such file or directory) while connecting to upstream, client: 127.0.0.1, server: , request: "GET /phpstatus HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "127.0.0.1"
2023-07-24 09:45:18,355 INFO success: child_exit_monitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-07-24 09:45:18,355 INFO success: php-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-07-24 09:45:18,355 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023-07-24 09:45:18,355 INFO success: mailhog entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2023/07/24 09:45:18 [crit] 2073#2073: *3 connect() to unix:/run/php-fpm.sock failed (2: No such file or directory) while connecting to upstream, client: 127.0.0.1, server: , request: "GET /phpstatus HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "127.0.0.1"
2023/07/24 09:45:19 [crit] 2075#2075: *5 connect() to unix:/run/php-fpm.sock failed (2: No such file or directory) while connecting to upstream, client: 127.0.0.1, server: , request: "GET /phpstatus HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "127.0.0.1"
[24-Jul-2023 09:45:19] NOTICE: fpm is running, pid 2069
[24-Jul-2023 09:45:19] NOTICE: ready to handle connections
[24-Jul-2023 09:45:19] NOTICE: systemd monitor interval set to 10000ms
============= contents of /mnt/ddev_config  =========
Error response from daemon: No such container: ddev-d9-db
======== Curl of site from inside container:
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 24 Jul 2023 09:45:23 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== curl -I of http://tryddevproject-8285.ddev.site from outside:
curl: (56) Recv failure: Connection reset by peer
Unable to curl the requested project Please provide this output in a new gist at gist.github.com.


```
</details>         

### Expected Behavior

The project should start normally

### Actual Behavior

The project fails with the error _"Failed to start app: Unable to listen on required ports, port 443 is already in use"_. Changing `router_https_port` to 8443 for example shifts the error to the next port. `ddev start` then produces the error _"Failed to start app: Unable to listen on required ports, port 8025 is already in use"_. After changing `mailhog_port` to 8300 the error changes to _"Failed to start app: Unable to listen on required ports, port 8026 is already in use"_. After changing `mailhog_https_port` to 8301 the error will simply move on to another port.
Checking the list of processes actively listening does not help. `lsof -i :443 -sTCP:LISTEN` for example does not display any processes. Neither does `netstat -tulnp | grep 443`.

### Steps To Reproduce

1. In Linux Ubuntu 20.04
2. Create minimal ddev project as suggested in [the troubleshooting docs](https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/#general-troubleshooting-strategies)
3. Run ddev start
4. See error "Failed to restart djh: Unable to listen on required ports, port 443 is already in use

**Comment by BobMali** [comment]
I followed all the steps which let me to the same conclusion as you that there is something wrong with the internet setup.
Because i wanted to start from scratch i tried to go through the entire ddev install process [for Linux](https://ddev.readthedocs.io/en/latest/users/install/ddev-installation/#manual). The first step doing `curl -fsSL https://pkg.ddev.com/apt/gpg.key | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/ddev.gpg > /dev/null` failed at the TLS-handshake. Opening https://pkg.ddev.com/ in a browser blocked the page with an error as well. However, using another device both curl and browser displayed no error. Since the device with the problem is a company device i contacted our tech support.
Long story short: I had to disable ipv6 on my device.

Entering 
```
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
```
in `/etc/sysctl.conf` and then applying `sudo sysctl -p` solved my problem.

Thanks again for your support and patience!

**Comment by rfay** [comment]
Glad you got it working! I hope you also get your DNS lookup working wrt 

> I also note the message "The hostname tryddevproject-8285.ddev.site is not currently resolvable", which means that you have something wrong with your internet setup, please see https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/#ddev-starts-but-browser-cant-access-url
